connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -115,7 +115,7 @@ public void initialize(TaskConfig taskConfig) {
     }
 
     protected void close() {
-        // nothing to do
+        producer.close();
```

Comment:
I was thinking about the same thing. The Connect code actually *never* calls `flush`. I've added the timeout. The choice of 30s was mostly arbitrary, but I don't want to expose a ton of configs just for timeouts like this.